PlaceRouteHierFlow/PnR-pybind11.cpp

```diff
@@ -313,6 +358,16 @@ PYBIND11_MODULE(PnR, m) {
   //m.def("save_state", &save_state, "helper function to save_state");
   //m.def("route_single_variant", &route_single_variant, "helper function to route a single variant");
   //m.def("route_top_down", &route_top_down, "helper function to perform top-down routing");
-  m.def("toplevel", &toplevel, "helper function to perform the whole C++ flow");
 
+  m.def("toplevel", [](const std::vector<std::string>& argv) {
+    py::scoped_ostream_redirect coutstream(
+        std::cout,
+        py::module_::import("align").attr("utils").attr("logging").attr("StreamLogger")(std::string("align.pnr.toplevel"), std::string("DEBUG"))
+    );
+    py::scoped_estream_redirect cerrstream(
+        std::cerr,
+        py::module_::import("align").attr("utils").attr("logging").attr("StreamLogger")(std::string("align.pnr.toplevel"), std::string("ERROR"))
+    );
```

Comment:
I have created a custom sink to redirect spdlog messages to the python logger so I feel like we no longer need to trap all cout and cerr messages. What do you think @stevenmburns ?

@854768750 the command line flag -l now modifies the C++ log-level as well. Almost all the messages (except the cout and cerr messages that are being trapped) will be shown under `align.pnr.default`. It is very easy to further specialize this by using `logger->error` instead of `spdlog::error` and so on where logger needs to be defined within each function (or class) as follows:
`  auto logger = spdlog::default_logger()->clone("CPPClassName.CPPFuncName");`
Should we just do this now or as a new PR? Should be doable as a giant `spdlog::` to `logger->` search replace followed by resolving compilation issues wherever 'logger' is not found.

I think this PR is ready to merge otherwise.

Reply:
> Should we just do this now or as a new PR? Should be doable as a giant spdlog:: to logger-> search replace followed by resolving compilation issues wherever 'logger' is not found.

Shouldn't take too long so working on this now.

Reply:
`auto logger = spdlog::default_logger()->clone("CPPClassName.CPPFuncName");`
Is this line copied in each cpp file?